File: etc/python/AddExample.py
# This example shows how database commands can be executed.
# Documentation: http://basex.org/api
#
# (C) BaseX Team 2005-11, ISC License

import BaseXClient, time

try:
  # initialize timer
  start = time.clock()

  # create session
  session = BaseXClient.Session('localhost', 1984, 'admin', 'admin')

  # perform command and print returned string
  print session.execute("create db testDB")
  print session.info()
  
  session.add("World.xml", "", "<x>Hello World!</x>")
  print session.info()

  # close session
  session.close()

  # print time needed
  time = (time.clock() - start) * 1000
  print time, "ms."

except IOError as e:
  # print exception
  print e

File: etc/php/AddExample.php
<?php
/*
 * This example shows how new documents can be added.
 * Documentation: http://basex.org/api
 *
 * (C) BaseX Team 2005-11, BSD License
 */
include("BaseXClient.php");

try {

  // create session
  $session = new Session("localhost", 1984, "admin", "admin");
  
  // create new database
  $session->execute("create db database");
  print $session->info();
  
  // add document
  $session->add("World.xml", "/world", "<x>Hello World!</x>");
  print "<br/>".$session->info();
  
  // add document
  $session->add("Universe.xml", "", "<x>Hello Universe!</x>");
  print "<br/>".$session->info();
  
  // run query on database
  print "<br/>".$session->execute("xquery /");
  
  // drop database
  $session->execute("drop db database");

  // close session
  $session->close();

} catch (Exception $e) {
  // print exception
  print $e->getMessage();
}
?>
